### PR TITLE
Fix dependabot issues

### PR DESCRIPTION
**Pull request by aleks-elkin**
Update many dependencies, e.g. `eslint` and it's plugins, `grunt` and it's plugins.
After upgrading eslint to a new major version I had to add a `@babel/core` and to change a code style to comply with new rules from newer versions of eslint and it's plugins.
Also, fixed vulnerability issues with these packages:
- `getobject` 
- `pac-resolver`
- `glob-parent`

### CODE REVIEW

File: tasks/browserstack.js [hunk 1]
```diff
@@ -122,6 +122,7 @@ const runr = function() {
  * Text: "Failure"
  * @see http://patorjk.com/software/taag/#p=display&f=Bloody&t=Failure
  */
+// eslint-disable-next-line no-redeclare
```
Comment (esmailpour-hosein) [inline tasks/browserstack.js:125]: I don't get why it should complain about redeclare, where it's being declared before?

File: spec/spiritual/spiritual-edb/arrays/edb.ArrayPopulator.spec.js [hunk 2]
```diff
@@ -24,10 +24,9 @@ describe('edb.ArrayPopulator', function likethis() {
 	});
 
 	it('should fail on bad input', function() {
-		var persons; // eslint-disable-line no-unused-vars
 		var fail = false;
 		try {
-			persons = new PersonCollection([false]);
+			var persons = new PersonCollection([false]); // eslint-disable-line no-unused-vars
```
Comment (esmailpour-hosein) [inline spec/spiritual/spiritual-edb/arrays/edb.ArrayPopulator.spec.js:29]: do we need to assign this to a variable?

**Review by esmailpour-hosein — APPROVED** [review]
🤞

File: tasks/browserstack.js (hunk 1 above)
Comment (aleks-elkin) [reply tasks/browserstack.js:125]: If we remove this assign, eslint will start to complain `error : Do not use 'new' for side effects no-new`
Which rule you want to disable? :)
I chose this one to introduce a minimal changes.

File: spec/spiritual/spiritual-edb/arrays/edb.ArrayPopulator.spec.js (hunk 2 above)
Comment (aleks-elkin) [reply spec/spiritual/spiritual-edb/arrays/edb.ArrayPopulator.spec.js:29]: If we remove this assign, eslint will start to complain `error : Do not use 'new' for side effects no-new`
Which rule you want to disable? :)

File: tasks/browserstack.js (hunk 1 above)
Comment (aleks-elkin) [reply tasks/browserstack.js:125]: You are right. Probably there was some mix-up with plugins (I updated not all of them in one go) and I had a false-positive on this line. Now it do not complain, so I will remove this.

File: spec/spiritual/spiritual-edb/arrays/edb.ArrayPopulator.spec.js (hunk 2 above)
Comment (esmailpour-hosein) [reply spec/spiritual/spiritual-edb/arrays/edb.ArrayPopulator.spec.js:29]: Then left it as it is :D I forgot that also something that makes es-lint complain

**Pull request merged**